src/server/stream_family.cc

```diff
@@ -603,6 +603,28 @@ int StreamTrim(const AddTrimOpts& opts, stream* s) {
   return 0;
 }
 
+class StreamMemTracker {
+ public:
+  StreamMemTracker() {
+    start_size_ = zmalloc_used_memory_tl;
+  }
+
+  void SetStreamSize(PrimeValue& pv) const {
+    const size_t current = zmalloc_used_memory_tl;
+    int64_t diff = static_cast<int64_t>(current) - static_cast<int64_t>(start_size_);
+    // If the diff is 0 it means the object use the same memory as before. No action needed.
+    if (diff == 0) {
+      return;
+    }
```

Comment:
Why is this needed? It doesn't seem like setting the same size has any performance implications, no?

Reply:
It's not, just me overengineering this condition when the function below covers it :)